WordBreak.py

```diff
@@ -0,0 +1,28 @@
+'''
+Word Break
+Given an input string and a dictionary of words,
+find out if the input string can be segmented into a space-separated sequence of dictionary words.
+For example, consider the following dictionary:
+{ pear, salmon, foot, prints, footprints, leave, you, sun, girl, enjoy },

+Examples:
Given the string “youenjoy”, 
+Output: True (The string can be segmented as “you enjoy”)

+Input: “youleavefootprints”,
+Output: True (The string can be segmented as “you leave footprints” or “you leave foot prints”)
+Input:salmonenjoyapples
+Output: False
+'''
+class Solution():
+    def wordBreak(self, s, dictionary):
+        if s == "" or dictionary == {}:
+            return False
+        s = s.lower()
+        for word in dictionary:
+            word = word.lower()
+            if word in s:
+                s = s.replace(word,"")
```

Comment:
If you had been given an input string "assomesort" and your dictionary contained the words ["some","assort"] your code would first replace "some" with "" and the new string would become "assort". The second run through would see that the word assort is in the string "assort" and the string would then be empty and return true. 
This case should return false though because we are not allowed to reshuffle words, but simply but spaces between them. This example should return true if the words in the dictionary were [as, some, sort].